### PR TITLE
fix: Upgrade testing library versions and resolve jest unit tests

### PR DESCRIPTION
**Pull request by Cal-L**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

What is included in this PR:

- Updates libraries to these versions:
`@testing-library/react": ^13.3.0 -> 14.0.0`
`@testing-library/react-native": ^11.5.0 -> 12.1.2`
`react-dom": 17.0.2 -> 18.2.0`
`react-test-renderer": 17.0.2 -> 18.2.0`

- Resolve all jest unit tests on the React Native 0.71.6 upgrade branch

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by Cal-L** [comment]
@SocketSecurity ignore-all

**Review by Fatxx — APPROVED** [review]
LGTM

File: app/util/test/testSetup.js [hunk 1]
```diff
@@ -269,6 +270,11 @@ jest.mock('react-redux', () => ({
...
 }));
 
+jest.mock('react-native-default-preference', () => ({
+  get: jest.fn(),
+  set: jest.fn(),
+}));
+
 // eslint-disable-next-line import/no-commonjs
 /* require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests(); */
 /* global.__reanimatedWorkletInit = jest.fn(); */
```
Comment (jpcloureiro) [inline app/util/test/testSetup.js:280]: Not related to your changes but we can replace these two lines with
```suggestion
require('react-native-reanimated/lib/module/reanimated2/jestUtils').setUpTests();
global.__reanimatedWorkletInit = jest.fn();
```

**Review by jpcloureiro — APPROVED** [review]
Changes LGTM 🚀 

Left a comment about reanimated jest setup we disabled before.

File: app/components/UI/FiatOnRampAggregator/components/Box.tsx [hunk 2]
```diff
@@ -47,8 +48,9 @@ const Box: React.FC<Props> = ({
   thin,
   onPress,
   activeOpacity,
-  accessible,
+  accessible = true,
   accessibilityLabel,
+  buttonTestID,
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/Box.tsx:53]: Should this me removed? If it needs to be kept I would do it conditionally on the presence of `onPress` like the `accessibilityRole` prop, wdyt?

**Pull request merged**